### ISSUE TITLE
"Get out of blueprint's way" constructor pathing issue

### ISSUE THREAD

**Issue by sprEEEzy** [beyond-all-reason/spring#1618]
### Description

Con building an Air lab.

https://github.com/user-attachments/assets/3d12017a-9918-4e27-bb1d-9a2be993709e

### Expected Behaviour

Moves top left.

### Actual Behaviour

Moves all the way down, because top side is blocked by con turrets?

### Reproduction steps

Cannot consistently reproduce. Encountered another bug which I'll present in another ticket.

**Comment by matthiaskrgr** [comment]
:laughing: probably related, I found it quite annoying that when you build winds in a tight grid for example, lets say

```
1 2 3 4 5 6 7 8 9
a b c d e f g h i k 
l m n o p q r s t u 
```
and the builder is standing at `3`, it then builds winds 1 and 2, moves to 4, builds 3, moves to 5, builds 4, etc ...

it would be nice if we could realize, "I'm building a batch here, so I'm moving to a spot where the next 4 queued buildings are reachable from it, but my position does not overlap with them"
